### PR TITLE
image hover component

### PR DESCRIPTION
**Pull request by nikitha2000**
image hover component

### CODE REVIEW

File: src/component/Body/GalleryLayout.jsx [hunk 1]
```diff
@@ -26,12 +28,24 @@ const Photos = () => {
     return <div>{error}</div>;
   }
 
+  const handleMouseEnter = (id) => {
+    setHoveredImageId(id);
+  };
+
+  const handleMouseLeave = () => {
+    setHoveredImageId(null);
+  };
```
Comment (vonnue-nidheesh) [inline src/component/Body/GalleryLayout.jsx]: You can handle the hovered state inside component itself.

File: src/component/Body/PhotoItem..css [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+  align-items: flex-end;
+  padding: 10px;
+  opacity: 0;
+}
+.top-left-button {
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem..css]: New line needed before

File: src/component/Body/GalleryLayout.css [hunk 3]
```diff
@@ -5,18 +5,6 @@
   padding: 20px;
```
Comment (vonnue-nidheesh) [inline src/component/Body/GalleryLayout.css:5]: Rename folder body to gallery/gallery layout.

File: src/component/Body/PhotoItem.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import "./PhotoItem..css";
+import Button from "../Button";
+
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: Sort imports

File: src/component/Body/PhotoItem..css [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  right: 20px;
+  background-color: #05a081;
+  color: white;
+  height: 50px;
+  width: auto;
+  border-radius: 100px;
+  border: none;
+  font-size: 16px;
+}
+.download-button img {
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem..css]: Newline

File: src/component/Body/PhotoItem.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const PhotoItem = ({ photo, hoveredImageId, onMouseEnter, onMouseLeave }) => {
+  return (
+    <div
+      className="photo-item"
+      onMouseEnter={() => onMouseEnter(photo.id)}
+      onMouseLeave={onMouseLeave}>
+      <img src={photo.thumbnailUrl} alt={photo.title} />
+
+      {hoveredImageId === photo.id && (
+        <div className="hover-buttons show">
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: Do you require classname `show` here?

File: src/component/Body/PhotoItem.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+          <div className="top-left-button">
+            <Button
+              label={<img src="/asset/save.svg" alt="Save Icon" />}
+              className="save-button"
+            />
+            <Button
+              label={<img src="/asset/heart.svg" alt="Heart Icon" />}
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: `alt="Favorite icon"`

File: src/component/Body/PhotoItem.jsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div
+      className="photo-item"
+      onMouseEnter={() => onMouseEnter(photo.id)}
+      onMouseLeave={onMouseLeave}>
+      <img src={photo.thumbnailUrl} alt={photo.title} />
+
+      {hoveredImageId === photo.id && (
+        <div className="hover-buttons show">
+          <div className="top-left-button">
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: Why the classname is `top-left-button`?

File: src/component/Body/PhotoItem.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            <Button
+              label={<img src="/asset/save.svg" alt="Save Icon" />}
+              className="save-button"
+            />
+            <Button
+              label={<img src="/asset/heart.svg" alt="Heart Icon" />}
+              className="save-button"
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: Rename class with `action-button`

File: src/component/Body/PhotoItem.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      onMouseEnter={() => onMouseEnter(photo.id)}
+      onMouseLeave={onMouseLeave}>
+      <img src={photo.thumbnailUrl} alt={photo.title} />
+
+      {hoveredImageId === photo.id && (
+        <div className="hover-buttons show">
+          <div className="top-left-button">
+            <Button
+              label={<img src="/asset/save.svg" alt="Save Icon" />}
+              className="save-button"
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: `action-button`

File: src/component/Body/PhotoItem.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            />
+            <Button
+              label={<img src="/asset/heart.svg" alt="Heart Icon" />}
+              className="save-button"
+            />
+          </div>
+          <div className="bottom-left">
+            <Button label={" Download"} className="download-button" />
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: `label=" Download"`

File: src/component/Body/PhotoItem.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+            <Button
+              label={<img src="/asset/save.svg" alt="Save Icon" />}
+              className="save-button"
+            />
+            <Button
+              label={<img src="/asset/heart.svg" alt="Heart Icon" />}
+              className="save-button"
+            />
+          </div>
+          <div className="bottom-left">
```
Comment (vonnue-nidheesh) [inline src/component/Body/PhotoItem.jsx]: `bottom-right-container`

Comment (vonnue-nidheesh) [reply src/component/Body/PhotoItem.jsx]: I think we missed showing the image title here.

File: src/component/Body/PhotoItem.jsx (hunk 6 above)
Comment (nikitha2000) [reply src/component/Body/PhotoItem.jsx]: no i removed it

**Review by vonnue-nidheesh — APPROVED** [review]

**Pull request merged**